model/src/pyrenew/process/ar.py

```diff
@@ -4,96 +4,84 @@
 from __future__ import annotations
 
 import jax.numpy as jnp
-import numpyro
-import numpyro.distributions as dist
-from jax import lax
 from jax.typing import ArrayLike
+from numpyro.contrib.control_flow import scan
 from pyrenew.metaclass import RandomVariable, SampledValue
+from pyrenew.process.iidrandomsequence import StandardNormalSequence
 
 
 class ARProcess(RandomVariable):
     """
-    Object to represent
-    an AR(p) process in
-    Numpyro
+    RandomVariable representing an
+    an AR(p) process.
     """
 
-    def __init__(
-        self,
-        name: str,
-        mean: float,
-        autoreg: ArrayLike,
-        noise_sd: float,
-    ) -> None:
+    def __init__(self, name, *args, noise_rv_suffix: str = "_noise", **kwargs):
         """
-        Default constructor
+        Default constructor.
 
         Parameters
         ----------
         name : str
-            Name of the parameter passed to numpyro.sample.
-        mean: float
-            Mean parameter.
-        autoreg : ArrayLike
-            Model parameters. The shape determines the order.
-        noise_sd : float
-            Standard error for the noise component.
+           A name for the process.
 
-        Returns
-        -------
-        None
+        noise_rv_suffix : str
+           A suffix to append to name when naming the
+           internal RandomVariable holding the process
+           noise. Default `"_noise"`.
         """
         self.name = name
-        self.mean = mean
-        self.autoreg = autoreg
-        self.noise_sd = noise_sd
+        super().__init__(*args, **kwargs)
+        self.noise_rv_ = StandardNormalSequence(name=name + noise_rv_suffix)
 
     def sample(
         self,
-        duration: int,
-        inits: ArrayLike = None,
+        n: int,
+        autoreg: ArrayLike,
+        init_vals: ArrayLike,
+        noise_sd: ArrayLike,
```

Comment:
Wouldn't this always be scalar?

Reply:
In the future we might want to enable a vector-valued `noise_sd` argument, but as the process is currently implemented, that wouldn't make sense. Changing the type hint accordingly.

Reply:
addressed in 3790b0c